engine/lite/runtime/src/test/scala/pl/touk/nussknacker/engine/lite/InterpreterTestRunnerTest.scala

```diff
@@ -102,4 +102,23 @@ class InterpreterTestRunnerTest extends AnyFunSuite with Matchers {
     results.externalInvocationResults("end") shouldBe List(ExternalInvocationResult("some-ctx-id", "end", 120))
   }
 
+  test("should handle fragment test parameters in test") {
+    val fragment = ScenarioBuilder
+      .fragment("fragment1", "in" -> classOf[String])
+      .filter("filter", "#in != 'stop'")
+      .fragmentOutput("fragmentEnd", "output", "out" -> "#in")
+
+    val parameterExpressions: Map[String, Expression] = Map(
+      "in" -> Expression("spel", "'some-text-id'")
+    )
+    val scenarioTestData = ScenarioTestData("fragment1", parameterExpressions)
+    val results = sample.test(fragment, scenarioTestData)
+
+    results.nodeResults("fragment1") shouldBe List(
+      NodeResult(ResultContext("fragment1", Map("in" -> "some-text-id")))
+    )
+    results.externalInvocationResults("fragmentEnd") shouldBe List(ExternalInvocationResult("fragment1", "fragmentEnd", Map("out" -> "some-text-id")))
```

Comment:
Why is it external invocation? I thought that external invocations are for services invocations tests. Why not nodeResults?

Reply:
Do you have an idea why it was done this way? I suspect that it was because someone wanted to see output of sink in unit tests - I mean you have sink with parameter `a: XYZ` which creates complex result where this a is somewhere used (e.g. nested). Am I right that we still also have those output fields as nodeResults (variables)? These externalInvocationResults are also visible in GUI?

Reply:
Ok, I tried to figure it out and it looks like in lite we have result in `nodeResults` `invocationResults` and `externalInvocationResults`. `externalInvocationResults` in this case is a result of invoking `SinkInvocationCollector` on sinks in `lite` mode. It is not used in sinks in flink (or at least not in this case) since in flink mode we have result only in `nodeResults` `invocationResults`.

I see it that way: 
- `nodeResults` - is a good place for it to be visible since we invoke noteEntered on node enter. 
- `invocationResults` - is also valid since its an effect on using `ExpressionEvaluator`
- `externalInvocationResults` - I am not sure but it can be omitted for sources in test. I've tried to remove it on two tests and I was able to replace it with check on `invocationResults`. I am nut sure yet if in all cases it will act the same but if you want me to I can try to remove this listener from sinks in test.

I've also updated tests to check both `nodeResults` `invocationResults`.

Reply:
Ok, I know now that `externalInvocationResults` is in lite tests for a reason - to collect output from sink after some transformation `LazyParamSink -> prepareResponse`. In tests it can be found here `pl.touk.nussknacker.engine.requestresponse.ParameterResponseSinkFactory.ParameterResponseSink`. In my case that does not matter but matters in different places. I've added it to my TODO list to try to figure it out and do some refactor. I will come back to it in another PR if that is ok with you.